pilot/pkg/proxy/envoy/v1/watcher.go

```diff
@@ -208,6 +222,54 @@ func watchCerts(ctx context.Context, certsDirs []string, watchFileEventsFn watch
 	watchFileEventsFn(ctx, fw.Event, minDelay, updateFunc)
 }
 
+// waitForCertsPresent is a blocking function that periodically checks the existance
+// of the certs, once every interval. It returns nil when all cert files are non-empty.
+// It returns error if the timeout is reached.
+func waitForCertsPresent(certs []CertSource, interval time.Duration, timeout time.Duration) (err error) {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	g, ctx := errgroup.WithContext(ctx)
+	g.Go(func() error {
```

Comment:
Why not check file persistence in parallel since the machinery is already in place?

Reply:
I see your point. I will update the code.